models/classes/QtiResultsService.php

```diff
@@ -354,4 +355,23 @@ private function getDisplayDate(string $epoch): string
     {
         return tao_helpers_Date::displayeDate($epoch, tao_helpers_Date::FORMAT_ISO8601);
     }
+
+
+    private function prepareItemVariableValue($value, $basetype): string
+    {
+        if ($basetype === 'file') {
+
+            if (extension_loaded('fileinfo')) {
+                $info = new finfo(FILEINFO_MIME_TYPE);
```

Comment:
should we use the file abstraction ?

Reply:
Here we have an access to binary file content only. Could you suggest in more details, how to do that?